Makefile

```diff
@@ -36,6 +36,13 @@ else
 export GOTESTCOMMAND=gotestsum --format pkgname --jsonfile testresults.json --
 endif
 
+# M1 Mac--homebrew install location in /opt/homebrew
+ifeq ($(OS_TYPE), darwin)
+ifeq ($(ARCH), arm64)
+export CPATH=/opt/homebrew/include
+export LIBRARY_PATH=/opt/homebrew/lib
```

Comment:
Is this something we should be setting here, or something that an M1 homebrew user should know needs to be in the shell environment already?

Reply:
I would argue against including this snippet, because strictly speaking homebrew has nothing to do with go-algorand and is not a dependency, and developers using homebrew on M1 will need to have this in their .bash_profile or .zprofile already. (?)

Reply:
I've taken a look, specifically https://github.com/Homebrew/brew/issues/9177 has a lot of discussion around this, and I'm not sure that I agree.

It's our script which is ultimately installing the boost library to this path via https://github.com/algorand/go-algorand/blob/master/scripts/configure_dev.sh#L80 and the homebrew developers have explicitly said that it's the tools which use brew's software which need to integrate with the changes (things which used to be automatically found by virtue of being in `/usr/` are now in `/opt/homebrew/` on M1).

Homebrew does provide `brew shellenv` to update the user's shell, but this is the output of that:
```
set -gx HOMEBREW_PREFIX "/opt/homebrew";
set -gx HOMEBREW_CELLAR "/opt/homebrew/Cellar";
set -gx HOMEBREW_REPOSITORY "/opt/homebrew";
set -q PATH; or set PATH ''; set -gx PATH "/opt/homebrew/bin" "/opt/homebrew/sbin" $PATH;
set -q MANPATH; or set MANPATH ''; set -gx MANPATH "/opt/homebrew/share/man" $MANPATH;
set -q INFOPATH; or set INFOPATH ''; set -gx INFOPATH "/opt/homebrew/share/info" $INFOPATH;
```
So it seems like it's not expected to be configured in your shell based on that.

Ultimately I'm fine with removing this, but it seems counterintuitive to me that we would `brew install boost` and then expect users to configure library variables themselves before running `make build`. 
Maybe we should be including these environment variable settings in `scripts/configure_dev.sh` instead of the makefile?

Reply:
I agree with Eric. The Makefile needs to ensure the environment is setup appropriately. In this case using `/opt/homebrew` is consistent with our configuration script. The only addition I would make here, is a comment explaining that this change is related to M1 and the new location where homebrew keeps binaries.